src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1428,10 +1431,10 @@ codet java_bytecode_convert_methodt::convert_instructions(
 
       assert(arg0.id()==ID_virtual_function);
 
-      // does the function symbol exist?
+      // does the function symbol exist? is it virtual and inherited?
```

Comment:
TBH I don't think the comment you suggest is conniving any additional information then the existing one.  With regards to, your second question, this is existing functionality, and we are simply trying to fix a bug, I don't know if there is better way to check if something is non-det but that would be out-of-the scope  of this pr (maybe, @smowton, can provide a better answer here) and is associated task